text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,495 @@
+3. If there are only successful outcomes, choose them in the following order:
+   `accepted`, `discarded`.
+
+### Outcomes Can Be Asymmetric
+
+There are a few reasons why a telemetry item may have a different outcome
+between incoming and outgoing item instruments.
+
+One example is when timeouts are configured independently between segments. A
+preceding segment's timeout being smaller than a following segment's timeout
+may cause outgoing items to have `timeout` outcomes while the corresponding
+incoming items have different outcomes.
+
+Another example is an OpenTelemetry Collector pipeline that is asynchronous.
+If a receiver responsible for recording the outcome of an incoming item is
+given a successful response by its following component, it will record an
+`accepted` outcome regardless of the result in, for example, the exporter. For
+these cases, exporters are expected to use the `deferred:` outcome categories
+to signal to monitoring systems especially the failure outcomes that were not
+seen by preceders.
+
+### SDK-specific considerations
+
+OpenTelemetry SDKs receive telemetry from multiple components. In
+addition to telemetry originating from OpenTelemetry APIs,
+signal-specific bridges also insert telemetry into the SDK pipeline.
+
+The asymmetry between incoming and outgoing item outcomes that is seen for
+Collectors is not a consideration for SDKs, because [OpenTelemetry
+APIs _by definition_ do not return errors to the
+caller](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md#api-and-minimal-implementation).
+The use of an incoming item metric instrument is not specified for SDKs,
+even though it can be meaningfully defined, because there is no additional 
+information gained from the metric. The origin of sdk_incoming_items is always 
+`inserted`, and the incoming item outcomes always match the outgoing item 
+outcomes in SDK pipelines. For this reason, there is no 
+`otelsdk_incoming_items` in this proposal.
+
+Similarly, SDKs are not meant to use the `deferred:` outcome prefix,
+as there is not a meaningful distinction to be made.
+
+Like the OpenTelemetry Collector, SDK pipelines may discard (e.g., by
+sampling) and drop (e.g., because queue-full) items.
+
+## Metric instrument summary
+
+The specified counter names are:
+
+- `otelcol_incoming_items`: The number of items received or inserted into a pipeline.
+- `otelcol_outgoing_items`: The number of items discarded, dropped, or exported by a Collector pipeline segment.
+- `otelsdk_outgoing_items`: The number of items discarded, dropped, or exported by a SDK pipeline segment.
+
+### Recommended conventional attributes
+
+- `otel.error` (boolean): This is true or false depending on whether the
+  outcome is considered a failure or a success.  See the chart below.
+- `otel.outcome` (string): This describes the outcome in a more specific
+  way than `otel.error`.
```

Comment:
@kristinapathak As mentioned at the SIG, I am interested in adding a similar attribute to the `otelcol_exporter_send_failed_*` metrics as part of https://github.com/open-telemetry/opentelemetry-collector/pull/10158 

I would be happy to use `outcome` if that is thought best.  

My one concern is that I see that attribute is used on another metric we use in our org, specifically the Micrometer generated `http.server.requests` metric, see [here for the ENUM values](https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/http/Outcome.java).  I see that this is **not** defined for the http [semconv](https://opentelemetry.io/docs/specs/semconv/http/http-metrics/), but I just thought it worth noting here for reference.

Reply:
I can see how `outcome` will be used by a variety of metrics. My hope is that with the `otel` prefix (ie `otel.outcome`) there is no conflict with the attribute name.

Do the outcome values defined here work to solve https://github.com/open-telemetry/opentelemetry-collector/issues/10157 or is more detail needed? If others are also in favor of this attribute, my hope is that you can update your PR to match this. 🙂